### PR TITLE
Feature/update cost calculation

### PR DESCRIPTION
**Pull request by TabeaMaria**
Fixes #Issue

**Changes proposed in this pull request**:
- new tests for cost calculation
- generalized cost calculation for balanced_market
- correction of flexible load
- some more small changes such as signs and docstrings

The following steps were realized (required):
- [x] Correct linting with `flake8 path_to_script`
- [x] Check if tests pass locally with `python -m pytest tests/`

Also the following steps were realized (if applies):
- [ ] Write docstrings to your code
- [ ] Explain new functionalities in readthedocs
- [x] Write test(s) for new patch of code
- [ ] Check building of documentation with `doc/make html`

### CODE REVIEW

**Comment by jakobgemassmer** [comment]
It looks like the strategy "distributed" is missing in line 106: 
`if (strategy == "greedy" or strategy == "balanced" or strategy == "distributed") \`

**Comment by jakobgemassmer** [comment]
The statements from line 112-120 and 121-129 are almost identical. Maybe shorten it by creating a string "<" or ">=" depending on the `utilization_time_per_year` and then putting this string into the statement, so that the statement for commodity and capacity costs only appears once?

**Comment by jakobgemassmer** [comment]
Typo in line 708. fixed -> fix: `# strategies without differentiation between fix and flexible load`

File: tests/test_calculate_costs.py [hunk 1]
```diff
@@ -123,6 +124,104 @@ def test_calculate_costs_advanced(self):
...
                 assert value == expected[i]
 
+    def test_calculate_costs_balanced_A(self):
+        scen_path = TEST_REPO_PATH / 'test_data/input_test_strategies/scenario_A.json'
+        with scen_path.open() as f:
+            j = json.load(f)
+        s = scenario.Scenario(j)
+        s.run('balanced', {"cost_calculation": True})
+        timeseries = s.GC1_timeseries
+        timeseries_lists = [timeseries.get(k, [0]*s.n_intervals) for k in [
```
Comment (jakobgemassmer) [inline tests/test_calculate_costs.py:134]:
```suggestion
        timeseries_lists = [timeseries.get(k, [0] * s.n_intervals) for k in [
```

**Review by jakobgemassmer — APPROVED** [review]

File: tests/test_calculate_costs.py (hunk 1 above)
Comment (TabeaMaria) [reply tests/test_calculate_costs.py:134]: done

**Pull request merged**